src/lib/autoinstall/clients/autoyast.rb

```diff
@@ -0,0 +1,196 @@
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "autoinstall/auto_sequence"
+
+Yast.import "Pkg"
+Yast.import "Wizard"
+Yast.import "Mode"
+Yast.import "Profile"
+Yast.import "AutoinstConfig"
+Yast.import "Y2ModuleConfig"
+Yast.import "Popup"
+Yast.import "AddOnProduct"
+Yast.import "CommandLine"
+Yast.import "AutoInstall"
+Yast.import "UI"
+
+module Y2Autoinstallation
+  module Clients
+    # This client is responsible for starting the AutoYaST UI.
+    class Autoyast < Yast::Client
+      include Yast::Logger
+
+      def initialize
+        textdomain "autoinst"
+
+        Yast::Mode.SetMode("autoinst_config")
+
+        Yast.include self, "autoinstall/dialogs.rb"
+        Yast.include self, "autoinstall/conftree.rb"
+      end
+
+      def main
+        if Yast::Y2ModuleConfig.GroupMap.empty?
+          Yast::Wizard.CreateDialog
+          Yast::Popup.Error(_("Error while reading configuration data."))
+          Yast::Wizard.CloseDialog
+          return :abort
+        end
+
+        turn_off_signature_checks
+
+        @cmdline = {
+          "id"         => "autoyast",
+          "help"       => _("AutoYaST"),
+          "guihandler" => fun_ref(method(:auto_sequence), "any ()"),
+          "actions"    => {
+            "file"         => {
+              "handler" => fun_ref(
+                method(:run_ui),
+                "boolean (map <string, string>)"
+              ),
+              "help"    => "file operations"
+            },
+            "module"       => {
+              "handler" => fun_ref(
+                method(:run_ui),
+                "boolean (map <string, string>)"
+              ),
+              "help"    => "module specific operations"
+            },
+            "list-modules" => {
+              "handler" => fun_ref(
+                method(:list_modules),
+                "void ()"
+              ),
+              "help"    => "list of known modules"
+            }
+          },
+          "options"    => {
+            "filename" => { "type" => "string", "help" => "filename=XML_PROFILE" },
+            "modname"  => { "type" => "string", "help" => "modname=AYAST_MODULE" }
+          },
+          "mappings"   => {
+            "file"   => ["filename", "modname"],
+            "module" => ["filename", "modname"]
+          }
+        }
+
+        ret = Yast::CommandLine.Run(@cmdline)
+        log.debug("ret = #{ret}")
+
+        Yast::AddOnProduct.CleanModeConfigSources
+        :exit
```

Comment:
this looks a bit strange, what is exit code?

Reply:
I am not sure, it was already there. But I will remove it because I think it is not needed.